packages/input/Input.tsx

```diff
@@ -91,30 +110,21 @@ export class Input extends Component<InputProps> {
     return [
       <style>{styles}</style>,
       <input
-        ref={this.setInerInputRef}
         className={className}
         type={type}
         value={value}
+        onInput={this.propagateOnChange}
         onChange={this.propagateOnChange}
-        onInput={this.setValue}
         placeholder={placeholder}
         disabled={disabled}
       />
     ];
   }
 
   private propagateOnChange( event: Event ) {
-    this.setValue();
-    emit( this, Input.events.change ); // emit change event on root element
+    // emit change event on root element
+    const input: Partial<HTMLInputElement> = event.target;
```

Comment:
we need to stop propagation of native event, otherwise our api would be leaky - this implementation will also trigger `onInput` regular DOM listener.


```ts
e.stopImmediatePropagation():
emit( this, Input.events.change, { detail: { data: input.value } });
```